client/src/modules/vouchers/voucher-form.service.js

```diff
@@ -180,7 +180,10 @@ function VoucherFormService(Vouchers, Constants, Session, VoucherItem, Cashboxes
         err = ERROR_SINGLE_ROW_TRANSACTION;
       }
 
-      var hasBalancedDebitsAndCredits = (this.totals.debit === this.totals.credit);
+      // validate that total debit equals to total credit
+      var totalDebit = Number(this.totals.debit).toFixed(2);
+      var totalCredit = Number(this.totals.credit).toFixed(2);
+      var hasBalancedDebitsAndCredits = (totalDebit === totalCredit);
```

Comment:
I worry that this will result in unbalanced transactions.  What happens if `14.1314 === 14.13`?  Will this be able to be posted to the General Ledger?

Reply:
With the development data set, we are in the impossibility to submit data for a voucher because `1.1885 !== 1.1885000000000001`

**For reproduce, define `Organisme Non Gouvernemental` as a convention.**

**Select all invoices of `Organisme Non Gouvernemental`**
![voucher convention select invoices](https://user-images.githubusercontent.com/5445251/30862482-17a7c5f6-a2c6-11e7-89b5-1092dcb5a9c8.png)

**The equality in the grid is there but debit and credit are not balanced**
![voucher convention imported](https://user-images.githubusercontent.com/5445251/30862535-4e37204e-a2c6-11e7-92f7-3af528a10dd6.png)

The real value of `this.totals.debit` = 1.1885
The real value of `this.totals.credit` = 1.1885000000000001
![real debit and credit](https://user-images.githubusercontent.com/5445251/30862602-86402814-a2c6-11e7-8fbf-7e39393fc907.png)

When we try to submit we have an error related to debit is not equal to credit.

Reply:
Our database truncates things at 4 decimal places, would it better to use this 4 decimal places?